src/cpu/aarch64/acl_gemm_convolution_utils.cpp

```diff
@@ -0,0 +1,285 @@
+            dst_tag = memory_desc_matches_one_of_tag(
+                    dst_md, nwc, nhwc, ncw, nchw);
+        }
+
+        if (acp.with_bias && bias_md.format_kind == format_kind::any)
+            CHECK(memory_desc_init_by_tag(bias_md, x));
+
+        is_nspc = utils::one_of(src_tag, nwc, nhwc);
+
+        memory_desc_t want_wei_md = weights_md;
+        auto wei_tag = is_nspc ? utils::pick(ndims - 3, wio, hwio)
+                               : utils::pick(ndims - 3, oiw, oihw);
+        CHECK(memory_desc_init_by_tag(want_wei_md, wei_tag));
+
+        // Compute Library does not support mismatching layouts
+        if ((src_tag != wei_tag) || (src_tag != dst_tag))
+            return status::unimplemented;
+
+        if (weights_md.format_kind == format_kind::any) {
+            weights_md = want_wei_md;
+        }
+        return (want_wei_md == weights_md) ? status::success
+                                           : status::unimplemented;
+    };
+
+    // TODO: look into changing default tag to the Compute Library default NHWC
+    auto default_dat_tag
+            = utils::pick(ndims - 3, format_tag::ncw, format_tag::nchw);
+    if (set_or_check_tags(default_dat_tag, default_dat_tag) != status::success)
+        return status::unimplemented;
+
+    const auto acl_layout = is_nspc ? arm_compute::DataLayout::NHWC
+                                    : arm_compute::DataLayout::NCHW;
+
+    // clang-format off
+    acp.src_info = arm_compute::TensorInfo(
+            arm_compute::TensorShape(iw, ih, ic, mb),
+            1,
+            arm_compute::DataType::F32,
+            acl_layout);
+
+    acp.wei_info = arm_compute::TensorInfo(
+            arm_compute::TensorShape(kw, kh, ic, oc),
+            1,
+            arm_compute::DataType::F32,
+            acl_layout);
+
+    acp.dst_info = arm_compute::TensorInfo(
+            arm_compute::TensorShape(ow, oh, oc, mb),
+            1,
+            arm_compute::DataType::F32,
+            acl_layout);
+
+    acp.bia_info = arm_compute::TensorInfo(
+            acp.with_bias ? arm_compute::TensorShape(oc)
+                          : arm_compute::TensorShape(),
+            1,
+            arm_compute::DataType::F32,
+            acl_layout);
+    // clang-format on
+
+    // Post-op activations
+    acp.act_info = acl_gemm_convolution_utils::get_acl_act(attr);
+
+    return status::success;
+}
+
+arm_compute::ActivationLayerInfo get_acl_act(const primitive_attr_t &attr) {
+    const auto &post_ops = attr.post_ops_;
+    const int entry_idx = post_ops.find(primitive_kind::eltwise);
+    if (entry_idx == -1) { return arm_compute::ActivationLayerInfo(); }
+
+    const auto eltwise_alg = post_ops.entry_[entry_idx].eltwise.alg;
+    float alpha = post_ops.entry_[entry_idx].eltwise.alpha;
+    float beta = post_ops.entry_[entry_idx].eltwise.beta;
+
+    using acl_act_t = arm_compute::ActivationLayerInfo::ActivationFunction;
+    acl_act_t acl_act_alg;
+    // clang-format off
```

Comment:
Yes, I tried to keep it minimal, in this case it happened that both the switch and the parameter list needed turning off so I kept it as a long segment instead of two shorter ones.
For me, the clang-format re-arranges the switch cases because of 'AllowShortCaseLabelsOnASingleLine: true' and it looks like [1]. I don't like it personally, but I'm happy to respect the convention if this is it.

I manually fixed the indentation mistake the sneaked in.

[1]
```
switch (eltwise_alg) {
    case eltwise_relu:
        // oneDNN defines RELU: f(x) = (x > 0) ? x : a*x
        // Compute Library defines LEAKY_RELU: f(x) = (x > 0) ? x : a*x
        // whilst Compute Library RELU is defined as: f(x) = max(0,x)
        if (alpha == 0) {
            acl_act_alg = acl_act_t::RELU;
        } else {
            acl_act_alg = acl_act_t::LEAKY_RELU;
        }
        break;
    case eltwise_tanh:
        // oneDNN defines TANH activation as:          f(x) = tanh(x)
        // Compute Library defines TANH activation as: f(x) = a*tanh(b*x)
        // Setting a=b=1 makes the two equivalent
        alpha = 1.f;
        beta = 1.f;
        acl_act_alg = acl_act_t::TANH;
        break;
    case eltwise_elu: acl_act_alg = acl_act_t::ELU; break;
    case eltwise_square: acl_act_alg = acl_act_t::SQUARE; break;
    case eltwise_abs: acl_act_alg = acl_act_t::ABS; break;
    case eltwise_sqrt: acl_act_alg = acl_act_t::SQRT; break;
    case eltwise_linear: acl_act_alg = acl_act_t::LINEAR; break;
    case eltwise_bounded_relu: acl_act_alg = acl_act_t::BOUNDED_RELU; break;
    case eltwise_soft_relu: acl_act_alg = acl_act_t::SOFT_RELU; break;
    case eltwise_logistic: acl_act_alg = acl_act_t::LOGISTIC; break;
    default: return arm_compute::ActivationLayerInfo();
    }
```

Reply:
I am fine with both approaches (though I think turning off clang-format for many switch statements is not very scalable, so unless manual formatting gives much better readability or ease of editing I would generally prefer avoid turn the switch off/on).

If you decide to stay with the current formatting, please put `// clang-format on` right after the switch ends, and yet another `// clang-format off` in the proper place in the next function. In other words, please make the chunks where clang format is disable the smallest possible, and do not cross the function or generally block boundaries. With the current state whatever function goes between `arm_compute::ActivationLayerInfo get_acl_act(const primitive_attr_t &attr)` and `bool acl_act_ok(alg_kind_t eltwise_activation)` will be with disabled clang-format.

Reply:
Ok, I decided to get rid of the entire block there.

Reply:
Thank you!
And sorry for nit picking.